helper/broker/notify_test.go

```diff
@@ -46,12 +46,12 @@ func TestGenericNotifier(t *testing.T) {
 	var notifiedWG sync.WaitGroup
 
 	for i := 0; i < 6; i++ {
-		go func(wg *sync.WaitGroup) {
-			wg.Add(1)
+		notifiedWG.Add(1)
+		go func() {
 			msg := notifier.WaitForChange(3 * time.Second)
 			require.Equal(t, "we got an update and not a timeout", msg)
-			wg.Done()
-		}(&notifiedWG)
+			notifiedWG.Done()
```

Comment:
Should this be in a defer so that a test failure doesn't leave the goroutine running?

Not a blocker, but this test would also benefit from some refactoring to eliminate the 500ms wait:
* Pass a buffered channel of `chan error` into the goroutine and have them send their errors out rather than potentially exiting the test inside a child goroutine.
* Call `Add` on the wait group in the outer goroutine and then have each inner goroutine call `Done` at the _top_, which would let us `wg.Wait` for them to start instead of waiting 500ms, and then poll on the `chan error` with a timeout.

Reply:
I added the defer but had some issues with the additional work. I'll merge this now, rather than spend additional time fiddling with that change and potentially revisit when I have more time.